src/app/core/common-components/json-editor/json-editor.component.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  ViewChild,
+} from "@angular/core";
+import { CommonModule } from "@angular/common";
+import {
+  Content,
+  ContentErrors,
+  createJSONEditor,
+} from "vanilla-jsoneditor/standalone.js";
+import { AlertService } from "../../alerts/alert.service";
+
+/**
+ * Component for editing JSON data.
+ */
+@Component({
+  selector: "app-json-editor",
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: "./json-editor.component.html",
+})
+export class JsonEditorComponent {
+  /**
+   * JSON value to be edited by the user.
+   */
+  @Input() value: object = {};
+
+  @Input() height = "65vh";
```

Comment:
can you design this component so that we can set any height through css on the `<app-json-editor style="height: 65vh"` and the json editor will fill whatever height it receives? That will make use much easier than having a custom input for the formatting here.

Reply:
Yes, I have updated the styling of the JSON editor component based on your suggestion. Now, by adding the height in the style from the parent component, we can manage the height of the JSON editor component directly from the parent. Please review the updated code and let me know if any changes are needed.

Reply:
Great :slightly_smiling_face: 

The width of the popup jumps from narrow to wide and back (when errors are displayed it becomes wide). Can you fix an good explicit width for the MatDialog (maybe some "80vw"? if it works well on both desktop and mobile)

Reply:
I have fixed this, and added the width for the `MatDialog` which works well on both desktop and mobile. Can you please check and let me know if any further changes are needed?

Reply:
works well now, thanks!